### PR TITLE
{WIP} Add push to MyGet from AppVeyor builds

### PR DESCRIPTION
**Pull request by erikbra**
Add a step to push to MyGet from the AppVeyor builds. Uses encryptet MyGet ApiKey (needs to be inserted, of course), using [https://ci.appveyor.com/tools/encrypt](https://ci.appveyor.com/tools/encrypt). It encrypts with a unique key per AppVeyor account. 

It would be really nice if AppVeyor supported merging of AppVeyor.yml and GUI, because then we could put the URI to the MyGet feed (which varies for e.g. erikbra and roundhouse) and the Api Key in the GUI instead of the checked-in file. But it does the job.

This only pushes "proper" builds (not builds of PRs). So we would push (of course with a -pre tag if #300 is merged too) whenever master is built on its own, but not on PR builds.

### CODE REVIEW

**Comment by erikbra** [comment]
I tried to register RoundhousE on MyGet, but it was already taken. (ref [https://www.myget.org/F/roundhouse](https://www.myget.org/F/roundhouse) exists), so I assumed some of you guys had it. @ferventcoder, do you have that account?

If not, is there any reason we don't just push to nuget.org?

**Comment by ferventcoder** [comment]
Looks like I own it. What is your account id?

**Comment by erikbra** [comment]
erikbra, as on github.

**Comment by ferventcoder** [comment]
Found you. Looks like you have access to it now as well.

**Comment by erikbra** [comment]
Lovely, thanks, @ferventcoder. You don't mind if I put the API token encrypter for use with AppVeyor in the source code?

**Comment by ferventcoder** [comment]
I'm not sure what you are asking, do you need the encrypted value for the API token?

**Comment by ferventcoder** [comment]
The encrypted value for the api token is `XWER/H0IkJ8JWFm+n3/lr7hHoNFAX/VpRD+AfpZo2EUgQkQKhrMxHa4aYZPogsvY`

**Comment by erikbra** [comment]
No, I can encrypt it myself with [https://ci.appveyor.com/tools/encrypt](https://ci.appveyor.com/tools/encrypt), it uses a per-AppVeyor-subscription key, I was just asking whether you were OK with med committing it (the encrypted value) to the repo. There shouldn't be any harm in doing so, no one else can decrypt it. I just wanted to double-check with you.

File: appveyor.yml [hunk 1]
```diff
@@ -2,12 +2,23 @@ version: 1.0.{build}
...
 - path: .\code_drop\log\*
-  name: Logs
\ No newline at end of file
+  name: Logs
+
+deploy:
+  provider: NuGet
+  server:  https://www.myget.org/F/roundhouse/api/v2/package
+  api_key:
+    secure: <add MyGet api key encrypted with https://ci.appveyor.com/tools/encrypt>
```
Comment (ferventcoder) [inline appveyor.yml]: `XWER/H0IkJ8JWFm+n3/lr7hHoNFAX/VpRD+AfpZo2EUgQkQKhrMxHa4aYZPogsvY` (just generated from the AV encrypt)

**Comment by ferventcoder** [comment]
@erikbra I believe it is per account, not user. In this case the chucknorris account is where you would need to be encrypting it, so you would be logging in under that github account.

**Comment by ferventcoder** [comment]
@erikbra if it comes out differently than what I posted here, then you may be under the wrong account.

**Comment by ferventcoder** [comment]
And yes, once encrypted it can be shared in the appveyor.yml file. That's the purpose of it. 👍

**Comment by ferventcoder** [comment]
@erikbra you have a merge issue, you should probably fetch upstream/master (this repo, assuming you have the name set as upstream) then do a rebase to your master. Follow that with a rebase against your branch from your local master. That will get you into the rebase conflict dialog so you can correct that.

**Comment by erikbra** [comment]
Thanks, @ferventcoder, I got a different value, but used the one you gave me. But it's not easy to test. Because on _my_ AppVeyor account, it fails, probably because it decrypts using my key. And the RoundhousE AppVeyor only tries to push on master branch. Should we merge and see if it works?

**Comment by ferventcoder** [comment]
We can't merge until you fix the merge conflicts and repush.

**Comment by erikbra** [comment]
Fixed the conflicts. Should we merge to see if it works with the key? Do you want to merge, @ferventcoder ? Or should I go ahead?

**Comment by ferventcoder** [comment]
Go ahead

**Comment by erikbra** [comment]
Yay, it worked! 

> Deploying using NuGet provider
> Publishing roundhouse.0.8.9-pre.1.nupkg to https://www.myget.org/F/roundhouse/api/v2/package...OK
> Publishing roundhouse.lib.0.8.9-pre.1.nupkg to https://www.myget.org/F/roundhouse/api/v2/package...OK
> Publishing roundhouse.msbuild.0.8.9-pre.1.nupkg to https://www.myget.org/F/roundhouse/api/v2/package...OK
> Total packages published: 3
> Build success

**Comment by BiggerNoise** [comment]
Awesome sauce!

**Pull request merged**